includes/class-taxjar-customer-record.php

```diff
@@ -183,117 +195,142 @@ public function get_from_taxjar() {
-			if ( empty( $first_name ) ) {
-				$first_name = $this->object->get_first_name();
-			}
-		}
-
-		if ( empty( $last_name ) ) {
-			$last_name = $this->object->get_billing_last_name();
-			if ( empty( $last_name ) ) {
-				$last_name = $this->object->get_last_name();
-			}
-		}
-
-		if ( empty( $first_name ) && empty( $last_name ) ) {
-			$name = '';
-		} else if ( empty( $first_name ) ) {
-			$name = $last_name;
-		} else if ( empty( $last_name ) ) {
-			$name = $first_name;
-		} else {
-			$name = $first_name . ' ' . $last_name;
-		}
-		$customer_data[ 'name' ] = $name;
-
-		$customer_data[ 'exempt_regions' ] = $this->get_exempt_regions();
+		$customer_data['customer_id']    = $this->get_customer_id();
+		$customer_data['name']           = $this->get_customer_name();
+		$customer_data['exemption_type'] = $this->get_exemption_type();
+		$customer_data['exempt_regions'] = $this->get_exempt_regions();
 
 		$country = $this->object->get_shipping_country();
 		if ( empty( $country ) ) {
 			$country = $this->object->get_billing_country();
 		}
 		if ( ! empty( $country ) ) {
-			$customer_data[ 'country' ] = $country;
+			$customer_data['country'] = $country;
 		}
 
 		$state = $this->object->get_shipping_state();
 		if ( empty( $state ) ) {
 			$state = $this->object->get_billing_state();
 		}
 		if ( ! empty( $state ) ) {
-			$customer_data[ 'state' ] = $state;
+			$customer_data['state'] = $state;
 		}
 
 		$postcode = $this->object->get_shipping_postcode();
 		if ( empty( $postcode ) ) {
 			$postcode = $this->object->get_billing_postcode();
 		}
 		if ( ! empty( $postcode ) ) {
-			$customer_data[ 'zip' ] = $postcode;
+			$customer_data['zip'] = $postcode;
 		}
 
 		$city = $this->object->get_shipping_city();
 		if ( empty( $city ) ) {
 			$city = $this->object->get_billing_city();
 		}
 		if ( ! empty( $city ) ) {
-			$customer_data[ 'city' ] = $city;
+			$customer_data['city'] = $city;
 		}
 
 		$address = $this->object->get_shipping_address();
 		if ( empty( $address ) ) {
 			$address = $this->object->get_billing_address();
 		}
 		if ( ! empty( $address ) ) {
-			$customer_data[ 'street' ] = $address;
+			$customer_data['street'] = $address;
 		}
 
 		$customer_data = apply_filters( 'taxjar_customer_sync_data', $customer_data, $this->object );
-		$this->data = $customer_data;
+		$this->data    = $customer_data;
 		return $customer_data;
 	}
 
 	/**
+	 * Retrieves the name of the customer.
+	 * Falls back to username if no shipping, billing or account names are available.
+	 *
+	 * @return string - Customer's name
+	 */
+	public function get_customer_name() {
+		$first_name = $this->object->get_shipping_first_name();
+		$last_name  = $this->object->get_shipping_last_name();
+
+		if ( empty( $first_name ) ) {
+			$first_name = $this->object->get_billing_first_name();
+			if ( empty( $first_name ) ) {
+				$first_name = $this->object->get_first_name();
+			}
+		}
+
+		if ( empty( $last_name ) ) {
```

Comment:
Will `first_name` every be initially empty while `last_name` is populated? Don't know that we'd want to run in to a scenario where we use the first name of one person and the last name of another.

Reply:
You're right, the logic here needed to be updated. I have changed it so it will only use the corresponding names (shipping, billing or account) together.